src/linux/address.rs

```diff
@@ -7,6 +7,28 @@ pub trait Ipv4AddrExt {
     fn from_address(sock: sockaddr) -> Self;
 }
 
+// Ext suffix is in case a dependency on an external MAC address crate is ever pulled in
+// Also matches above naming this way
+pub trait MACAddrExt {
+    fn to_address(&self) -> sockaddr;
+    fn from_address(sock: sockaddr) -> Self;
+}
+
+pub struct MACAddr {
+    data: [u8; 6],
+}
+impl MACAddr {
+    pub fn new(data: [u8; 6]) -> Self {
+        Self { data }
+    }
+}
+impl Into<[u8; 6]> for MACAddr {
```

Comment:
please replace it with:

```rust
impl From<MacAddr> for [u8; 6] {

}
```

> One should avoid implementing [Into](https://doc.rust-lang.org/std/convert/trait.Into.html) and implement [From](https://doc.rust-lang.org/std/convert/trait.From.html) instead. Implementing [From](https://doc.rust-lang.org/std/convert/trait.From.html) automatically provides one with an implementation of [Into](https://doc.rust-lang.org/std/convert/trait.Into.html) thanks to the blanket implementation in the standard library.

https://doc.rust-lang.org/std/convert/trait.Into.html